src/qt/treectrl.cpp

```diff
@@ -56,275 +495,699 @@ void wxTreeCtrl::SetIndent(unsigned int indent)
+        ? wxQtConvertTreeItem(qTreeItem->child(0))
+        : wxTreeItemId();
 }
 
-wxTreeItemId wxTreeCtrl::GetNextChild(const wxTreeItemId& item, wxTreeItemIdValue& cookie) const
+wxTreeItemId wxTreeCtrl::GetNextChild(
+    const wxTreeItemId& item,
+    wxTreeItemIdValue& cookie
+) const
 {
+    wxCHECK_MSG(item.IsOk(), wxTreeItemId(), "invalid tree item");
+
+    int currentIndex = reinterpret_cast<int>(cookie);
+    ++currentIndex;
+
+    const QTreeWidgetItem *qTreeItem = wxQtConvertTreeItem(item);
+
+    if ( currentIndex < qTreeItem->childCount() )
+    {
+        cookie = reinterpret_cast<wxTreeItemIdValue>(currentIndex);
+        return wxQtConvertTreeItem(qTreeItem->child(currentIndex));
+    }
+
     return wxTreeItemId();
 }
 
 wxTreeItemId wxTreeCtrl::GetLastChild(const wxTreeItemId& item) const
 {
-    return wxTreeItemId();
+    wxCHECK_MSG(item.IsOk(), wxTreeItemId(), "invalid tree item");
+
+    const QTreeWidgetItem *qTreeItem = wxQtConvertTreeItem(item);
+    const int childCount = qTreeItem->childCount();
+    return childCount == 0 
+        ? wxTreeItemId()
+        : wxQtConvertTreeItem(qTreeItem->child(childCount - 1));
 }
 
 wxTreeItemId wxTreeCtrl::GetNextSibling(const wxTreeItemId& item) const
 {
-    return wxTreeItemId();
+    wxCHECK_MSG(item.IsOk(), wxTreeItemId(), "invalid tree item");
+
+    QTreeWidgetItem *qTreeItem = wxQtConvertTreeItem(item);
+    QTreeWidgetItem *parent = qTreeItem->parent();
+
+    if ( parent != NULL )
+    {
+        int index = parent->indexOfChild(qTreeItem);
+        wxASSERT(index != -1);
+
+        ++index;
+        return index < parent->childCount()
+            ? wxQtConvertTreeItem(parent->child(index))
+            : wxTreeItemId();
+    }
+
+    int index = m_qtTreeWidget->indexOfTopLevelItem(qTreeItem);
+    wxASSERT(index != -1);
+
+    ++index;
+    return index < m_qtTreeWidget->topLevelItemCount()
+        ? wxQtConvertTreeItem(m_qtTreeWidget->topLevelItem(index)) 
+        : wxTreeItemId();
 }
 
 wxTreeItemId wxTreeCtrl::GetPrevSibling(const wxTreeItemId& item) const
 {
-    return wxTreeItemId();
+    wxCHECK_MSG(item.IsOk(), wxTreeItemId(), "invalid tree item");
+
+    QTreeWidgetItem *qTreeItem = wxQtConvertTreeItem(item);
+    QTreeWidgetItem *parent = qTreeItem->parent();
+
+    if ( parent != NULL )
+    {
+        int index = parent->indexOfChild(qTreeItem);
+        wxASSERT(index != -1);
+
+        --index;
+        return index >= 0
+            ? wxQtConvertTreeItem(parent->child(index))
+            : wxTreeItemId();
+    }
+
+    int index = m_qtTreeWidget->indexOfTopLevelItem(qTreeItem);
+    wxASSERT(index != -1);
+
+    --index;
+    return index >= 0
+        ? wxQtConvertTreeItem(m_qtTreeWidget->topLevelItem(index))
+        : wxTreeItemId();
 }
 
 wxTreeItemId wxTreeCtrl::GetFirstVisibleItem() const
 {
+   wxTreeItemId itemid = GetRootItem();
+    if ( !itemid.IsOk() )
+        return itemid;
+
+    do
+    {
+        if (IsVisible(itemid))
```

Comment:
```suggestion
        if ( IsVisible(itemid) )
```

Reply:
Fixed. I even went through all of these last night. I guess I missed a few. :)